src/explorer/CosmosDBTreeItem.ts

```diff
@@ -38,20 +38,38 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
                 label: 'Install Cosmos DB Extension...'
             })];
         }
-        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
-        const connections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
-        // tslint:disable-next-line:strict-boolean-expressions
-        const unit = connections.find((x: IConnections) => x.webAppId === this.root.client.id) || <IConnections>{};
-        if (!unit.cosmosDB || unit.cosmosDB.length === 0) {
+
+        const connectionStrings: string[] = [];
+        const appSettingsNode = this.parent.parent.appSettingsNode;
+        const allAppSettings = await appSettingsNode.ensureSettings();
+        if (allAppSettings.properties) {
+            const dictionary: { [key: string]: string } = allAppSettings.properties;
+            Object.keys(dictionary).forEach((key) => {
+                connectionStrings.push(dictionary[key]);
+            });
+        }
+
+        const allConnectionItems = connectionStrings.map(connectionId => {
+            try {
```

Comment:
Avoid using a try/catch where you ignore errors. You might accidentally ignore the wrong errors. I suggest something like this:

```typescript
const treeItems: CosmosDBDatabase[] = [];
for (const connectionString of connectionStrings) {
    if (<connectionString is a mongo connection string>) {
        treeItems.push(new CosmosDBDatabase(...));
    }
}
...
return treeItems;
```